lib/kubernetes-deploy/rollout_conditions.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class RolloutConditionsError < StandardError
+  end
+
+  class RolloutConditions
+    class << self
+      def parse_conditions(conditions_string)
+        return default_conditions if conditions_string.downcase == "true"
+
+        conditions = JSON.parse(conditions_string).deep_symbolize_keys
+        errors = validate_conditions(conditions)
+        raise RolloutConditionsError, errors.join("\n") unless errors.empty?
+
+        # Create JsonPath objects
+        conditions[:success_conditions].each do |query|
+          query[:path] = JsonPath.new(query[:path]) if query.key?(:path)
+        end
+        conditions[:failure_conditions].each do |query|
+          query[:path] = JsonPath.new(query[:path]) if query.key?(:path)
+          query[:error_msg_path] = JsonPath.new(query[:error_msg_path]) if query.key?(:error_msg_path)
+        end
+
+        conditions
+      rescue RolloutConditionsError
+        raise
+      rescue JSON::ParserError => e
+        raise RolloutConditionsError, "Error parsing rollout conditions: #{e}"
+      rescue RuntimeError => e
+        raise RolloutConditionsError,
+          "parse_conditions encountered an unknown error." \
+          "This is most likely caused by an invalid JsonPath expression. Failed with: #{e}"
+      end
+
+      def default_conditions
+        {
+          success_conditions: [
+            {
+              path: JsonPath.new('$.status.conditions[?(@.type == "Ready")].status'),
+              value: "True",
+            },
+          ],
+          failure_conditions: [
+            {
+              path: JsonPath.new('$.status.conditions[?(@.type == "Failed")].status'),
+              value: "True",
+              error_msg_path: JsonPath.new('$.status.conditions[?(@.type == "Failed")].message'),
+            },
+          ],
+        }
+      end
+
+      private
+
+      def validate_conditions(conditions)
+        errors = []
+
+        top_level_keys = [:success_conditions, :failure_conditions]
+        missing = top_level_keys.reject { |k| conditions.key?(k) }
+        unless missing.blank?
+          errors << "Missing required top-level key(s): #{missing}"
```

Comment:
Wouldn't it be valid for them to provide success conditions but not failure conditions? That's the case for a lot of core resource types we've manually modelled.